flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageMemoryManagerImpl.java

```diff
@@ -91,6 +94,15 @@ public class TieredStorageMemoryManagerImpl implements TieredStorageMemoryManage
      */
     private final Map<Object, Integer> numOwnerRequestedBuffers;
 
+    /**
+     * The queue that contains all available buffers. This field should be thread-safe because it
+     * can be touched both by the task thread and the netty thread.
+     */
+    private final BlockingQueue<MemorySegment> bufferQueue;
```

Comment:
How is the thread safety achieved?

Reply:
The `BlockingQueue` implementations are required to be thread-safe according to the Java API doc.